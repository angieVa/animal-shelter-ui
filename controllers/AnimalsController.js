import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            url: `animals`,
            baseURL: process.env.API,
            data: animal,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
              },
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
}

