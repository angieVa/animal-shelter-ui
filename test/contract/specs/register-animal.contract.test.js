import { provider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

const animal = {
    name: "Jasper",
    breed: "Bicolor",
    gender: "Male",
    vaccinated: false
}

describe('Given An Animal service', () => {
    describe('When a request to add a new animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'new animal',
                uponReceiving: 'a request to add a new animal',
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    headers: {
                        'Content-Type': 'application/json',
                      },
                    body: Matchers.like(animal),
                },
                willRespondWith: {
                    status: 201,
                    body: {
                        name: Matchers.string("Jasper"),
                        breed: Matchers.string("Bicolor"),
                        gender: Matchers.string("Male"),
                        vaccinated: Matchers.boolean(true),
                        },
                    headers: {
                        'Content-Type': 'application/json',
                      },
                }
            });
        });

        it("Then it should return the right data", async() =>{

            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
}); 