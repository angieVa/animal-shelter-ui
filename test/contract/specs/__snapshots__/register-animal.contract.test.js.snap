// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal service When a request to add a new animal is made Then it should return the right data 1`] = `
Object {
  "breed": "Bicolor",
  "gender": "Male",
  "name": "Jasper",
  "vaccinated": true,
}
`;
